src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```
@@ -351,4 +351,16 @@
     <ProjectCapability Include="RoslynComponent" Condition="'$(IsRoslynComponent)' == 'true'"/>
   </ItemGroup>
 
+  <!--
+    ========================
+    SourceGeneratorProjectReference Support
+    ========================
+  -->
+  <Target Name="TranslateSourceGeneratorProjectReferenceItems"
+          BeforeTargets="ResolveProjectReferences">
+    <ItemGroup>
+      <ProjectReference Include="@(SourceGeneratorProjectReference)" OutputItemType="Analyzer" ReferenceOutputAssembly="false" AdditionalProperties="%(SourceGeneratorProjectReference.AdditionalProperties);IsRidAgnostic=true" />
```

`ReferenceOutputAssembly` should still be a supported property. You'll need to use an item definition group to default this value to `false`.

```suggestion
      <ProjectReference Include="@(SourceGeneratorProjectReference)" OutputItemType="Analyzer" ReferenceOutputAssembly="%(ReferenceOutputAssembly)" AdditionalProperties="%(SourceGeneratorProjectReference.AdditionalProperties);IsRidAgnostic=true" />
```